discovery/gossiper.go

```diff
@@ -1872,17 +1876,24 @@ func (d *AuthenticatedGossiper) processRejectedEdge(
 		msg:    chanAnn,
 	})
 	if e1Ann != nil {
-		announcements = append(announcements, networkMsg{
-			source: d.selfKey,
-			msg:    e1Ann,
-		})
+		// Only add the ChanUpdate to the gossiper if the
+		// __dont_forward__ bit signals it.
+		if !e1Ann.MessageFlags.HasDontForward() {
+			announcements = append(announcements, networkMsg{
+				source: d.selfKey,
+				msg:    e1Ann,
+			})
+		}
 	}
 	if e2Ann != nil {
-		announcements = append(announcements, networkMsg{
-			source: d.selfKey,
-			msg:    e2Ann,
-		})
-
+		// Only add the ChanUpdate to the gossiper if the
+		// __dont_forward__ bit signals it.
+		if !e2Ann.MessageFlags.HasDontForward() {
```

Comment:
I believe in this case since we are processing a rejected edge, the only case where this happens is one where these flags are set, or even if they aren't, they ought to be.

Reply:
We don't know for sure, depending on how strict the peer rate-limits us, but in the best case scenario we should have the same message bits set here. I think it makes sense to check for the bit nonetheless just in case we somehow did not update the Graph-Data and do not broadcast a ChanUpdate which was never met to be broadcasted.

Reply:
This code seems fine as-is, but can you elaborate -- what does the peer rate-limiting have to do with this?

Reply:
So I was referring to the rate limiting because we consider a ChanUpdate as stale if it has not a greater timestamp than the prior one. (even if the content differ). This had effects in the itests where blocks were mined  quickly and the time between block 3 and block 6 was lower than 1 sec, and when we or our peer announced the channel after 6 blocks it had happened that the old ChanUpdate from our Peer was still in our db. That should never happen in the real world but thats why this check is also important here (at least for the itests).